### DISCUSSION TITLE
How to test that it actually works?

### DISCUSSION THREAD

**Discussion by tomvdp** [hako-mikan/sd-webui-regional-prompter#260] [General]
I was suspecting there must be something wrong with my setup as I cannot control the regions in a picture. So I thought of this simple test:

> a blue dot on a plain (white background) ADDCOL
> a green dot on a plain (white background) ADDROW
> an orange dot on a plain (white background) ADDCOL
> a black dot on a plain (white background)

The Divide Ratio is set to 1,1,1;1,1,1  in Columns mode on a square image. (i.e. 4 equal squares)

It just doesn't generate 4 colored dots. I mostly get a single colored dot on a grey background.

I am curious what would be the outcome of this test on someone else's setup.
Am I correct to assume it should generate 4 dots?
What would be a better test strategy?

**Comment by Symbiomatrix** [comment]
Regional is more of a modification of details in a single image, it doesn't run multiple generations and stitch them together like multidiffusion.
1) What you asked for here is essentially a single dot, which in each corner is coloured differently. Putting it in the center of the image is an expected default behaviour for most models.
2) If you got only a single colour or a mix of the colours, that's probably attention mode - it does a sort of compression which means regions blend together better. Latent has a more clear cut separation and will tend to allow higher variance on regions (but again, you're fighting against most models' training data with chimeras).
3) You placed both the background colour and the dot colour in the same region, which means the context is gonna mix those colours up somewhat. Furthermore, unless the model understands and favours all white backgrounds (which most SD1.5 variants don't), you won't get all white backgrounds most of the time.

As for a better strategy:
1) You'll want to put "multiple dots" or somesuch in the common clause, in addition to the colour and details in regions. It still won't be accurate with the number and spread, not much to do about that other than using a lora / mask mode trickery for superfine positioning (like marking a circular region and saying "(((((blue)))))" etc).
2) White background goes into base clause (with lots of emphasis on white). Use a low base weight in 2D mode so it doesn't affect the subjects, or use mask mode with a weight of 0 so it just fills the gap regions.

**Reply by tomvdp** [reply]
Thank you for your elaborate answer. I was trying to do something impossible anyway... I discovered it is just impossible to even ask for a single dot. Well, you can ask, but it will never generate a single dot. Weird that such a simple thing is too difficult for all models I tried.
